Tasks/Ignatchik/Additional_task/Task_1/search_and_close_tag.py

```diff
@@ -0,0 +1,156 @@
+import re
+
+with open("input.html", 'r', encoding='utf-8') as file:
+    list_html = file.readlines()
+
+flag = True
+html_line = ""
+list_of_all_tags = []
+pattern = '<(\S*?)[^>]*>.*?|<.*?/>'
+
+with open("input.html", "r") as input_file:
+    for line in input_file:
+        html_line += line
+    
+all_tags = re.finditer(pattern, html_line)
+
+try:
+    while flag:
+        item = next(all_tags)
+        list_of_all_tags.append(item.group())
+except StopIteration:
+    flag = False
+
+single_HTML_tags = ['!DOCTYPE', 'area', 'base', 'br', 'col', 'embed', 'hr', 'img', 'input', 'keygen', 'link', 'meta', 'param', 'source', 'track', 'wbr']
```

Comment:
it's better to use a tuple in such cases, this collection should not change during the program execution